crates/node/src/validate.rs

```diff
@@ -414,7 +417,8 @@ impl StateRead for State {
                 }
                 Ok(values)
             })
-            .await?
+            .await
+            .map_err(StateReadError::from)
         }
         .boxed()
```

Comment:
I think we should rollback these changes to the `key_range` method as this changes the behaviour quite a bit, and also no-longer supports the dry-run case. This section of `validate.rs` is a bit different from the other `conn_pool` use-cases as it aims to also support "dry-runs".